payments-core/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/TransformToPaymentMethodCreateParams.kt

```diff
@@ -54,15 +53,16 @@ internal class TransformToPaymentMethodCreateParams {
         // Need to convert Country Fields to a country code to put it in the parameter map
         val formKeyValueMap = formFieldValues.fieldValuePairs
             .mapValues { entry ->
-                when (entry.key) {
-                    Country.identifier -> {
-                        transformCountry(entry.value)
+                val fieldEntry = entry.value
+                when (fieldEntry.type) {
+                    ElementType.Country -> {
+                        transformCountry(fieldEntry.value)
```

Comment:
Can this be a method in `ElementType`, which is a no-op by default and transforms `Country` and `IdealBank`? Then the `when` can go away.

Reply:
I am more inclined to move it into the Transformation to FormFieldValues and the Populate from FormFieldValues, thoughts?

@brnunes-stripe

Reply:
SGTM, I was just thinking if we could get rid of the `when`.

Reply:
Have the country and ideal conversion happen in the TransformToFormFieldValues with the help of the Config.

This meant:
- adding a rawField to the controller which would map the initial value to the raw one with the help of the Controller.kt
- adding a function to convert a raw value into a display value with the help of the controller.